server.js

```diff
@@ -24,12 +31,210 @@ const app = express();
+It defines the fields for each document, including their data types and any validation rules.
+
+We define each field in the schema using a JSON object with the field name as the key and its data type as the value.
+For example, the "Character Name" field has a data type of String. Additionally, 
+we use the Number data type for the "Character ID" field.
+
+After defining the schema, we create a Mongoose model for the Characters collection using mongoose.model(). 
+The first argument is the name of the collection, which is "Characters" in our case. 
+The second argument is the schema that defines the structure of the documents in the collection.
+
+We also create a spellsSchema and a Mongoose model for the Spells collection in a similar fashion. 
+This schema specifies the structure of the documents that will be stored in the Spells collection. 
+The schema includes fields such as "Incantation", "Spell Name", and "Effect", with their respective data types.
+
+These schemas and models will be used later in the code to perform *CRUD operations on the database.
+*/
+
+/*
+In the code block below the schema definitions, there is a conditional block of code that checks for a RESET_DATABASE environment variable. 
+If this variable is present, 
+it will delete all data from the Characters and Spells collections in the MongoDB database, 
+and then insert all the characters and spells from the characters.json and spells.json files into the respective collections. 
+This is useful for resetting the database to a known state during development or testing.
+*/
+if (process.env.RESET_DATABASE) {
+  const resetDatabase = async () => {
+    await Characters.deleteMany();
+    await Spells.deleteMany();
+    charactersPotter.forEach((singleCharacter) => {
+      const newCharacter = new Characters(singleCharacter);
+      newCharacter.save()
+    });
+    spellsPotter.forEach((singleSpell) => {
+      const newSpell = new Spells(singleSpell)
+      newSpell.save()
+    })
+  }
+  resetDatabase();
+}
+
+
+
 // Start defining your routes here
 app.get("/", (req, res) => {
-  res.send("Hello Technigo!");
+  const navigation = {
+    guide: "Routes for Harry Potter API",
+    Endpoints: [
+      {
+        "/characters": "Display all Characters from Harry Potter Movies",
+        "/characters/ID/:ID": "Search specific Character id",
+        "/characters/name/:name": "Search for a name in Harry Potter Movies",
+        "/spells": "Display all spells"
+      },
+    ],
+  };
+  res.send(navigation);
 });
 
+
+
+app.get("/characters", async (req, res) => {
+  try {
+    res.json(charactersPotter)
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: "Server error" })
+  }
+})
+
+app.get("/spells", async (req, res) => {
+  try {
+    res.json(spellsPotter);
+  } catch (err) {
+    console.error(err);
+    res.status(500).json({ message: "Server error" })
+  }
+});
+
+
+app.get("/characters/ID/:ID", async (req, res) => {
```

Comment:
I am not able to fetch character by ID. Not sure if the param entered is wrong.
This was my url:
`https://project-mongo-api-pb7rmnzmyq-lz.a.run.app/characters/ID/2`
It throws this error:
`"Cast to ObjectId failed for value \"2\" (type string) at path \"_id\" for model \"Characters\"`

Reply:
Ok great, I changed it now. The reason the user couldn't find the character by using ID/(ex 1 for Harry) is that the endpoint I had defined is looking for the MongoDB _id field, not the Character ID field. The findById method in Mongoose is designed to find documents by their unique _id field, not other fields in the document.
To allow users to search for a character by their Character ID, I needed to modify the endpoint to use the findOne method in Mongoose, which allows you to search by any field in the document.
With this modification, users will be able to search for a character by using their Character ID field, for example, https://project-mongo-api-pb7rmnzmyq-lz.a.run.app/characters/ID/1 (Harry Potter)